### PR TITLE
Add tests for `Gabow`, `WordGraph` and `Paths`

### PR DESCRIPTION
**Pull request by MTWhyte**
This PR adds some tests for the mentioned stuff, which seemed to get lost in some earlier merging.

### CODE REVIEW

**Comment by MTWhyte** [comment]
I've updated the tests, deleting some stuff that isn't used.

There are a couple of things I've noticed that I'll ask about when convenient, e.g. about how to 'unset' the 'source' node in the `Paths` object.

File: tests/test_paths.py [hunk 1]
```diff
@@ -32,7 +32,6 @@ def test_001():
         w.set_target(i, i % 2, i + 1)
 
     p = Paths(w)
-    assert p.at_end()
```
Comment (james-d-mitchell) [inline tests/test_paths.py]: Just to say that I think this is a bug in libsemigroups, that the return value of `at_end()` is non-deterministic if the Paths object isn't properly initialised (it somehow isn't tracked in the class whether or not it is initialised, but it is implicitly assumed that it is).

**Comment by james-d-mitchell** [comment]
Am I ok to merge this @MTWhyte ?

**Comment by MTWhyte** [comment]
Yeah go ahead!

On 17 Apr 2024, at 16:28, James Mitchell ***@***.***> wrote:

﻿

Am I ok to merge this @MTWhyte<https://github.com/MTWhyte> ?

**Pull request merged**